EPUB loading optimizations, support obfuscated embedded fonts

poire-z opened the pull request:
#### LVContainer: optimize filename lookups with a hashtable

In LVNamedContainer, and so for its subclasses like directories and ZIP archives, maintain and use a name to index table when looking and opening files by name.

#### EPUB: optimize OPF parsing

- Optimize EpubItems id lookups with a hashtable.
- Avoid using doc-`>nodeFromXPath(..[n])` n times, as it is slow, and just iterate the children nodes instead.
- Also add some comments.
- Optimize metadataOnly handling, and tweak it so we get a single exit point.

(I have not tested all the toc & pagemap types, because I can't easily find the documents that has each type, so I hope I didn't mess any - but the changes are all similar.)

Some big book (3000 items in the zip) I have used over the years for benchmarking (and that I remember took 70 seconds loading at some point in the past), was taking these days 64 seconds. With this PR and #493, it now takes 48 seconds on my Kobo.

#### EPUB: support obfuscated embedded fonts

There is code to demangle fonts obfuscated by the Adobe algorithm, but it has never been triggered for 10+ years because, although `META-INF/encryption.xml` was parsed, its items where not stored and the list of encrypted items stayed empty!
So, properly handle that, and also add support for the alternative algorithm from the IDPF/EPUB specs (which needs us to compute a sha1 of the book identifier, so add a `sha1digest()` function).
The "Reading of DRM protected books is unsupported" message was also never shown: show it now when we notice all items are encrypted with an unsupported algorithm.

#### Fonts/CSS: handle "oblique" as "italic"

Also  parse `font-style: oblique` inside `@font-face {}`, as not doing it may get the font associated to "regular"

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/koreader/crengine/495)
<!-- Reviewable:end -->

poire-z commented:
@benoit-pierre : I made these little optimizations, which may impact your benchmarks results. I'm curious if they do, and how much with each of your methods for estimating page numbers.
They may need little changes to your crengine modifications (I also tweaked the metadataOnly bits, so you may have to tweak them).

benoit-pierre commented:
We'll look into it. One thing I found that really speeds up things on my Kindle is to [avoid calculating the document CRC32](https://github.com/koreader/crengine/commit/74e7f90887d94b84bd9062b72433fcc053de109e) (since it's only used by the cache): it's almost twice as fast loading ~1GB of documents for metadata only.

poire-z commented:
@benoit-pierre : one thing I would have mentionned later - but I might as well do it now: you're currently focusing on EPUBs, but crengine/koreader should handle as well the other filetypes (fb2, html, txt, otf, docx, chm...) So, your tweaks, if they end up going in, will need to be adapted/extended for the other file types (ie. these should also get their crc32 computed - looks like you removed all of them :)

benoit-pierre commented:
I know, probably just track all calls to `openFromCache`.

poire-z commented:
Added support for obfuscated fonts in EPUBs.
The code to demangle Adobe-obfuscated was already there (but not used).
Adding support for the other obfuscated algorithm was mostly just adding sha1 computation and doing similarly to the Adobe algo (the IDPF/EPUB people probably just stole the idea, and decided to add unnecessary sha1 to that, because _we are who we are, we made the awful EPUB3 specs, why stop there_ :)
I only have 5 + 5 books with each of these algo among my 400+ test EPUBs, so it's not really popular, but well... (@Frenzie: the _Dead trees give no shelter_ epub you once sent me has some nice IDPF-obfuscated fonts - and it had the issue with `font-style: oblique` I fixed, that we didn't see until now as the obfuscated fonts were ignored).

Frenzie commented:
Very neat  :-)
